src/App.tsx

```diff
@@ -9,36 +9,178 @@ import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
 
+import { User } from './types/User';
+import { Post } from './types/Post';
+import { Errors } from './types/Errors';
+import { Comment } from './types/Comment';
+
+import {
+  getComments,
+  getPosts,
+  getUsers,
+} from './api/apiActions';
+
 export const App: React.FC = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [selectedUserId, setSelectedUserId] = useState(0);
+  const [isDropDownActive, setIsDropDownActive] = useState(false);
+  const [selectedUsersPosts, setSelectedUsersPosts] = useState<Post[]>([]);
+  const [isLoadingPosts, setIsLoadingPosts] = useState(false);
+  const [errorMessage, setErrorMessage] = useState<Errors>(Errors.noError);
+  const [isLoadingComments, setIsLoadingComments] = useState(false);
+  const [comments, setComments] = useState<Comment[] | null>(null);
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const [isFormVisible, setIsFormVisible] = useState(false);
+
+  const toggleDropDown = () => {
+    setIsDropDownActive(!isDropDownActive);
+  };
+
+  const closeDropDown = () => {
+    setIsDropDownActive(false);
+  };
+
+  const toggleCommentForm = () => {
+    setIsFormVisible(!isFormVisible);
+  };
+
+  const selectDropDownItem = (user: User, userId: number) => {
+    setSelectedUser(user);
+    setSelectedUserId(userId);
+    setIsDropDownActive(false);
+    setSelectedPost(null);
+    setComments(null);
+  };
+
+  const selectPost = (post: Post) => {
+    setSelectedPost(post);
+    if (isFormVisible) {
+      toggleCommentForm();
+    }
+  };
+
+  const deSelectPost = () => {
+    setSelectedPost(null);
+  };
+
+  const extendComments = (commentToPost: Comment) => {
+    if (!comments) {
+      setComments([commentToPost]);
+    } else {
+      setComments([...comments, commentToPost]);
+    }
```

Comment:
it is better to avoid `if/else` where it is possible
```suggestion
  const postComments = comments.length > 0
    ? [...comments, commentToPost]
    :  [commentsToPost];
    
  setComments(postComments);
```